component/http/component.go

```diff
@@ -190,6 +192,18 @@ func (cb *Builder) WithWriteTimeout(wt time.Duration) *Builder {
 	return cb
 }
 
+// WithShutdownGracePeriod sets the Shutdown Grace Period for the HTTP component.
+func (cb *Builder) WithShutdownGracePeriod(gp time.Duration) *Builder {
```

Comment:
We should provide a unit test for this also.

Reply:
There is a test here for all of them - `Test_createHTTPServerUsingBuilder` when I've added this setter and nothing for each of parameters. Do we need to create a specific only for this one?

Reply:
Added